SpatialGDK/Source/SpatialGDK/Private/Utils/ActorGroupManager.cpp

```diff
@@ -0,0 +1,202 @@
+#include "Utils/ActorGroupManager.h"
+#include "SpatialGDKSettings.h"
+
+UActorGroupManager::UActorGroupManager()
+{
+	InitFromSettings();
+}
+
+UActorGroupManager* UActorGroupManager::GetInstance()
+{
+	UActorGroupManager* Manager = NewObject<UActorGroupManager>();
+	Manager->InitFromSettings();
+	return Manager;
+}
+
+void UActorGroupManager::InitFromSettings()
+{
+	const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>();
+
+	ClassPathToActorGroup.Empty();
+	ActorGroupToWorkerType.Empty();
+
+	auto ActorGroups = DefaultActorGroups();
+	auto WorkerAssociation = DefaultWorkerAssociation();
+	if (Settings->bEnableOffloading)
+	{
+		ActorGroups = Settings->ActorGroups;
+		WorkerAssociation = Settings->WorkerAssociation.ActorGroupToWorker;
+	}
+
+	for (TPair<FName, FActorClassSet> ActorGroup : ActorGroups)
+	{
+		for (TSoftClassPtr<AActor> ClassPtr : ActorGroup.Value.ActorClasses)
+		{
+			ClassPathToActorGroup.Add(ClassPtr, ActorGroup.Key);
+		}
+	}
+
+	ActorGroupToWorkerType.Append(WorkerAssociation);
+}
+
+FName UActorGroupManager::GetActorGroupForClass(UClass* Class)
+{
+	if (Class == nullptr || !Class->IsChildOf<AActor>())
+	{
+		return NAME_None;
+	}
+
+	UClass* FoundClass = Class;
+	TSoftClassPtr<AActor> ClassPtr = TSoftClassPtr<AActor>(FoundClass);
+
+	while (FoundClass != nullptr && !ClassPathToActorGroup.Contains(ClassPtr))
+	{
+		FoundClass = FoundClass->GetSuperClass();
+		ClassPtr = TSoftClassPtr<AActor>(FoundClass);
+	}
+
+	if (FoundClass != nullptr)
+	{
+		if (ClassPathToActorGroup.Contains(ClassPtr))
+		{
+			FName ActorGroup = ClassPathToActorGroup.FindRef(ClassPtr);
+			ClassPathToActorGroup.Add(TSoftClassPtr<AActor>(Class), ActorGroup);
+			return ActorGroup;
+		}
+	}
+	return NAME_None;
+}
+
+FName UActorGroupManager::GetWorkerTypeForClass(UClass* Class)
+{
+	FName ActorGroup = GetActorGroupForClass(Class);
+
+	if (ActorGroupToWorkerType.Contains(ActorGroup))
```

Comment:
```suggestion
	if (const FName* WorkerType = ActorGroupToWorkerType.Find(ActorGroup))
	{
	     return *WorkerType;
	}
```

Reply:
Not sure, but wouldn't the suggested change allow us to only do one lookup here?